src/repo-service.cpp

```diff
@@ -138,3 +295,132 @@ void RepoService::openLocalFile(const QString& repo_id,
         helper->start();
     }
 }
+
+void RepoService::onGetRequestSuccess(const ServerRepo& repo)
+{
+    GetRepoRequest* req = qobject_cast<GetRepoRequest*>(sender());
+    if (!req)
+        return;
+
+    SyncedSubfolderhasRepoID subfolder_helper(repo.id);
+    std::vector<SyncedSubfolder>::iterator pos = std::find_if(synced_subfolders_.begin(), synced_subfolders_.end(), subfolder_helper);
+
+    if (pos != synced_subfolders_.end()) {
+        ServerRepo fixed_repo = repo;
+        fixed_repo.parent_path = pos->parent_path();
+        fixed_repo.parent_repo_id = pos->parent_repo_id();
+        server_repos_.push_back(fixed_repo);
+    }
+
+    delete req;
+    get_repo_reqs_.pop_front();
+
+    if (!get_repo_reqs_.empty())
+        get_repo_reqs_.front()->send();
+    else
+        emit refreshSuccess(server_repos_);
```

Comment:
The code between line (315, 321) is the same as (330, 336), why not refactor them to a common subroutine?